sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExporterBuilder.java

```diff
@@ -351,10 +361,43 @@ private SpanExporter buildTraceExporter(ConfigProperties configProperties) {
     private MetricExporter buildMetricExporter(ConfigProperties configProperties) {
         HeartbeatExporter.start(
             MINUTES.toSeconds(15), createDefaultsPopulator(configProperties), builtTelemetryItemExporter::send);
+        if (connectionString != null) {
+            StatsbeatModule statsbeatModule = new StatsbeatModule(PropertyHelper::lazyUpdateVmRpIntegration);
+            statsbeatModule.start(
+                builtTelemetryItemExporter,
+                this::getStatsbeatConnectionString,
+                connectionString::getInstrumentationKey,
+                false,
+                getStatsbeatInterval(STATSBEAT_LONG_INTERVAL_SECONDS, configProperties.getInt(STATSBEAT_LONG_INTERVAL_SECONDS)),
+                getStatsbeatInterval(STATSBEAT_SHORT_INTERVAL_SECONDS, configProperties.getInt(STATSBEAT_SHORT_INTERVAL_SECONDS)),
+                false,
+                initStatsbeatFeatures());
+        }
         return new AzureMonitorMetricExporter(
             new MetricDataMapper(createDefaultsPopulator(configProperties), true), builtTelemetryItemExporter);
     }
 
+    private long getStatsbeatInterval(String key, Integer intervalFromConfig) {
+        if (intervalFromConfig == null) {
+            if ("STATSBEAT_LONG_INTERVAL_SECONDS".equals(key)) {
+                return DAYS.toSeconds(1); // default every 24 hours
+            } else if ("STATSBEAT_SHORT_INTERVAL_SECONDS".equals(key)) {
+                return MINUTES.toSeconds(15); // default every 15 mins
+            }
+        }
+        assert intervalFromConfig != null;
+        return SECONDS.toSeconds(intervalFromConfig);
+    }
+
+    private Set<Feature> initStatsbeatFeatures() {
+        // TODO (jean): start tracking native image usage based on a system property or env var to indicate it's from the native image path
```

Comment:
Thanks! I imagine that it is not only specific to native images? For example, the features used by the Application Insights Java agent will have to be returned by this method?

Reply:
I imagine that an enum value (for example `GRAAL_VM_NATIVE`) will have to be added to the `Feature` enum to track native iamges?

Reply:
yes, feature name can be flexible as long as we can identify native image.